### PR TITLE
cmd,client,daemon: expose "force devmode" in sysinfo

### PR DESCRIPTION
**Pull request by zyga**
The GNOME software center would like to show appopriate icons if a snap
application is confined and while each snap shows its confinement
information the whole system may be unable to offer effective
confinement. This patch adds a "confinement" key to the sysinfo
interface and switches the existing "snap debug confinement" command to
use it, rather than implement the logic in the client.

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

**Comment by robert-ancell** [comment]
Looks good, thanks!

File: daemon/api.go [hunk 1]
```diff
@@ -267,6 +267,16 @@ func sysInfo(c *Command, r *http.Request, user *auth.UserState) Response {
 			Next:     formatRefreshTime(nextRefresh),
 		},
 	}
+	// NOTE: Right now we don't have a good way to differentiate if we
+	// only have partial confinement (ala AppArmor disabled and Seccomp
+	// enabled) or no confinement at all. Once we have a better system
+	// in place how we can dynamically retrieve these information from
+	// snapd we will use this here.
+	if release.ReleaseInfo.ForceDevMode() {
+		m["confinement"] = "none"
```
Comment (chipaca) [inline daemon/api.go]: is it really "none", or is it "only seccomp"?

Comment (zyga) [inline daemon/api.go]: It's none for the POV of the user. We don't want to drill down to the "this or that" details.

Comment (Conan-Kudo) [inline daemon/api.go]: I don't know if it's a great idea to make it like that. It's a more negative lie than the current one. Can we say something like "basic" or "full" confinement?

**Comment by robert-ancell** [comment]
This field is an enum right, rather than a free-form string? (Asking for documentation and snapd-glib implementation purposes).

File: daemon/api.go (hunk 1 above)
Comment (zyga) [inline daemon/api.go]: We can evolve this but I just kept the values that we have already.

**Comment by robert-ancell** [comment]
This is required for Fedora because without this change GNOME Software shows all snaps as confined. As this is not true in Fedora they want this information so snaps are correctly marked as not confined (the default for RPM packages).

File: daemon/api.go (hunk 1 above)
Comment (mvo5) [inline daemon/api.go]: The latest idea was "partial" or "strict"

**Comment by robert-ancell** [comment]
Anything blocking this other than the CI build?

**Comment by zyga** [comment]
@robert-ancell I think it's just CI now, I think tests will pass now, the Linode problem was fixed yesterday by Gustavo.

**Comment by zyga** [comment]
I fixed the tests, it should go green soon.

**Pull request merged**